File: code/src/main/java/it/unifi/nave/data/EventContainer.java
package it.unifi.nave.data;

import java.util.HashMap;
import java.util.Map;

public class EventContainer implements Hashable {
  private String author;
  private Map<String, byte[]> mapOfKeys;
  private byte[] payload;

  public EventContainer(String author) {
    this.author = author;
    this.mapOfKeys = new HashMap<>();
  }

  public String getAuthor() {
    return author;
  }

  public void setAuthor(String author) {
    this.author = author;
  }

  public Map<String, byte[]> getMapOfKeys() {
    return mapOfKeys;
  }

  public void addKey(String id, byte[] key) {
    mapOfKeys.put(id, key);
  }

  public byte[] getPayload() {
    return payload;
  }

  public void setPayload(byte[] payload) {
    this.payload = payload;
  }
}

File: UniBlock/src/main/scala/data/EventContainer.scala
package it.unifi.nave.uniblock
package data

import data.event.Event.EventType

// TODO Differenziare i container per i certificati
case class EventContainer(author: String, eventType: EventType, payload: String, sign: String) extends Hashable {
  private var _mapOfKeys: Map[String, String] = Map.empty

  def addKey(id: String, key: String): Unit = {
    addKey(id -> key)
  }

  def addKey(key: (String, String)): Unit = {
    _mapOfKeys += key
  }

  def mapOfKeys: Map[String, String] = _mapOfKeys

  override def toString: String = {
    s"""hash = $hash
       |author = $author
       |eventType = $eventType
       |payload = $payload
       |sign = $sign
       |$mapOfKeysToString""".stripMargin
  }

  private def mapOfKeysToString: String = mapOfKeys.map(f => s"Receiver = ${f._1} -> Key = ${f._2}").mkString("\n")
}
